src/panel/direction/RouteResult.jsx

```diff
@@ -34,11 +34,21 @@ export default class RouteResult extends React.Component {
     });
   }
 
+  componentDidUpdate(prevProps) {
+    if (this.props.routes.length !== prevProps.routes.length) {
```

Comment:
This is not really related to this feature but a fix to prevent an error when going from a vehicle mode to another one with less alternatives. We explicitly reset the selected route to the first to avoid having an active route index greater than the number of routes.